app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -38,10 +38,18 @@ def connection_params(host)
       params = {
         'ansible_port' => host_port(host),
         'ansible_user' => host_user(host),
-        'ansible_ssh_pass' => host_ssh_pass(host),
+        'ansible_become' => host_become(host),
         'ansible_connection' => connection_type(host),
         'ansible_winrm_server_cert_validation' => winrm_cert_validation(host)
       }
+      # Check if private_key is defined, if it is use that. Otherwise
+      # use ssh_pass
+      if !host.host_params['ansible_ssh_private_key_file'].empty? ||
+          !Setting[:private_key].empty?
```

Comment:
I think this should be `Setting[:ansible_ssh_private_key_file]`. Note that condition will be true out of the box since you set the default value to `/tmp/private_key_file`, so I would recommend keeping it blank. You can also take advantage of your `host_private_key_file` method and change it to:

```ruby
unless host_private_key_file(host).empty?
# ...
else
# ...
end
```